Source/RxSwift/partition.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  partition.swift
+//  RxSwiftExt
+//
+//  Created by Shai Mishali on 24/11/2018.
+//  Copyright © 2018 RxSwiftCommunity. All rights reserved.
+//
+
+import RxSwift
+
+public extension ObservableType {
+    /**
+     Partition a stream into two separate streams of elements that match, and don't match, the provided predicate.
+
+     - parameter predicate: A predicate used to filter matching and non-matching elements.
+
+     - returns: A tuple of two streams of elements that match, and don't match, the provided predicate.
+    */
+    func partition(_ predicate: @escaping (E) throws -> Bool) -> (matches: Observable<E>, nonMatches: Observable<E>) {
+        let stream = self.share()
```

Comment:
Just wondering, because I often do the same in my custom operators, is it really fine to `share()` an observable under the hood? It might be an unexpected side effect don't you think?

Reply:
I think it's the common path - no "harm" done from doing that for the common use cases. I can't really imagine a use case where you wouldn't want to share though. 


Usually I do agree that it's a bit weird to use it under the hood but for something like this, I feel it makes sense. If there are any issues with this we can expand it later on, with some flag etc.